### ISSUE TITLE
No Release file for Ubuntu 24.04 LTS

### ISSUE THREAD

**Issue by aklehm** [apache/couchdb#5077]
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description
I want to install CouchDB on a Ubuntu Server 24.04 LTS and followed the offical docs here: [CouchDB docs](https://docs.couchdb.org/en/latest/install/unix.html#enabling-the-apache-couchdb-package-repository)

But when update the repositores with `sudo apt update`, I get the following error messages.

```shell
Hit:1 http://ports.ubuntu.com/ubuntu-ports noble InRelease
Hit:2 http://ports.ubuntu.com/ubuntu-ports noble-updates InRelease
Hit:3 http://ports.ubuntu.com/ubuntu-ports noble-backports InRelease
Hit:4 http://ports.ubuntu.com/ubuntu-ports noble-security InRelease
Ign:5 https://apache.jfrog.io/artifactory/couchdb-deb noble InRelease
Err:6 https://apache.jfrog.io/artifactory/couchdb-deb noble Release
  404   [IP: 44.226.59.123 443]
Reading package lists... Done
E: The repository 'https://apache.jfrog.io/artifactory/couchdb-deb noble Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

## Steps to Reproduce

Fresh installation of Ubuntu Server 24.04 LTS
```shell
sudo apt update
sudo apt upgrade
```
**Enabling the Apache CouchDB package repository**
```shell
sudo apt install -y curl apt-transport-https gnupg
curl https://couchdb.apache.org/repo/keys.asc | gpg --dearmor | sudo tee /usr/share/keyrings/couchdb-archive-keyring.gpg >/dev/null 2>&1
source /etc/os-release
echo "deb [signed-by=/usr/share/keyrings/couchdb-archive-keyring.gpg] https://apache.jfrog.io/artifactory/couchdb-deb/ ${VERSION_CODENAME} main" | sudo tee /etc/apt/sources.list.d/couchdb.list >/dev/null
sudo apt update
```

## Expected Behaviour

Update repositories und install CouchDB after enabling the Apache CouchDB package repository

## Your Environment

Hardware: Raspberry Pi 3B
OS: Ubuntu Server 24.04 LTS

* CouchDB version used: latest
* Browser name and version: -
* Operating system and version: Ubuntu Server 24.04 LTS

**Comment by big-r81** [comment]
Hi,

Ubuntu 24.04 LTS isn't listed under the [convenience binary packages](https://docs.couchdb.org/en/latest/install/unix.html#installation-using-the-apache-couchdb-convenience-binary-packages) section in our docs, so you get the error message you encountered. I think and hope that this will be done in the near future.

In the meantime you could use the [Snap package](https://snapcraft.io/couchdb) or build CouchDB on your Raspi by yourself.

**Comment by goobemaster** [comment]
@aklehm Docs does not explicitly explain that you have to either set this env variable or replace ${VERSION_CODENAME} to a OS version code name, in my case it was "bookworm" for Debian.

Try running "lsb_release -cs". If you get a correct code name you could do:

```shell
echo "deb [signed-by=/usr/share/keyrings/couchdb-archive-keyring.gpg] https://apache.jfrog.io/artifactory/couchdb-deb/ ${lsb_release -cs} main" | sudo tee /etc/apt/sources.list.d/couchdb.list >/dev/null
``` 

But if you've run the command already fix the issue by manually editing /etc/apt/sources.list.d/couchdb.list

**Comment by alleycat58uk** [comment]
Is there an ETA for adding a convenience package for Ubuntu 24.04 LTS?

**Comment by DarrenCook** [comment]
Is the "not-a-bug" tag what is stopping this getting sorted?

We want to use the official package on Ubuntu24.04 LTS, but can't.

We tried using the "jammy" version with:
 `echo "deb [signed-by=/usr/share/keyrings/couchdb-archive-keyring.gpg] https://apache.jfrog.io/artifactory/couchdb-deb/ jammy main" | sudo tee /etc/apt/sources.list.d/couchdb.list >/dev/null`

But that gives:

```
The following packages have unmet dependencies:
 couchdb : Depends: libmozjs-78-0 (>= 78.15.0) but it is not installable
      Depends: libicu70 (>= 70.1-1~) but it is not installable
```

**Comment by dilevembamu12** [comment]
hello brother , this how to install couchdb to ubuntu 24.04 TLS (Noble)
STEP 1 : download manually de package for ubuntu "noble" version

`cd /tmp`

`wget https://ci-couchdb.apache.org/job/jenkins-cm1/job/FullPlatformMatrix/job/main/lastSuccessfulBuild/artifact/pkgs/noble/couchdb_3.4.2-e137b72%7Enoble_amd64.deb`


 STEP 2: Install manually de package by fix dependencies in the same way

`sudo apt-get --fix-broken install ./couchdb_3.4.2-e137b72~noble_amd64.deb`

STEP 3: Fill forms in the configuration popup 

STEP 4 : Run and check status
`systemctl start couchdb`
`systemctl status couchdb`

STEP 5: optional , check the database connection
`curl http://127.0.0.1:5984/`

**Comment by DarrenCook** [comment]
BTW, I thought this had been fixed and closed on Dec 14, but I see  https://docs.couchdb.org/en/latest/install/unix.html#installation-using-the-apache-couchdb-convenience-binary-packages still doesn't list Ubuntu 24.04.

It looks like the deb now exists, so maybe just the documentation needs updating and this can be closed?

**Comment by dilevembamu12** [comment]
Yes indeed, the deb is now available on the official Apache couchdb site. But apparently until today there is no official documentation for this

**Comment by Jay-Lokhande** [comment]
else you can compile it from the source, it ran on my ubuntu 24.04 system

**Comment by big-r81** [comment]
This is resolved now, packages for noble exists, documentation for the "stable" version are updated, but we need to backport this into the "main" docs when doing the 3.5.0 release @nickva ...
